testing/tests/test_runtime.py

```diff
@@ -118,7 +121,8 @@ def test_env_clean_on_charm_error():
             assert 'JUJU_REMOTE_APP' in os.environ
             _ = 1 / 0  # raise some error
     # Ensure that some other error didn't occur (like AssertionError!).
-    assert 'ZeroDivisionError' in str(exc.value)
+    exc = exc_info.value if bare_charm_errors else exc_info.value.__cause__
+    assert isinstance(exc, ZeroDivisionError)
```

Comment:
+1 on the discussion in the other comment that the code flow should not depend on the value of the test parameter.

Then most tests could read something like:

```suggestion
    assert isinstance(exc, ZeroDivisionError) or isinstance(exc_info.value.__cause__, ZeroDivisionError)
```